### PR TITLE
Change `stc_fwd` to use RFP instead of packed format

### PR DESCRIPTION
**Pull request by jbadger95**
RFP format has similar storage requirements to packed format but can use BLAS 3 operations; I don't recall the timings off the top of my head (it's been a while since I did this) but I believe this was around 30% faster.

### CODE REVIEW

File: trunk/src/modules/stc.F90 [hunk 1]
```diff
@@ -342,31 +342,31 @@ subroutine stc_fwd_herm(Ni,Nb, Aii,Abi,Aib,Abb,Bi,Bb)
    VTYPE  , intent(inout) :: Aii(Ni,Ni),Abi(Nb,Ni),Aib(Ni,Nb),Abb(Nb,Nb), &
                              Bi(Ni,NR_RHS),Bb(Nb,NR_RHS)
 !
-!..workspace for Hermitian bubble matrix in packed format
+!..workspace for Hermitian bubble matrix in RFP format
    VTYPE, dimension(Nb*(Nb+1)/2) :: AP
 !
-   integer  :: i,j,k,nk,k1,k2
+   integer  :: i,j,k,k1,k2
```
Comment (stefanhenneking) [inline trunk/src/modules/stc.F90:348]:
```suggestion
```

File: trunk/src/modules/stc.F90 [hunk 2]
```diff
@@ -342,31 +342,31 @@ subroutine stc_fwd_herm(Ni,Nb, Aii,Abi,Aib,Abb,Bi,Bb)
...
+   stop
+endif
 !
 !..Cholesky factorization
 #if C_MODE
-   call ZPPTRF('U', Nb, AP, info)
+   call ZPFTRF('N','U', Nb, AP, info)
 #else
-   call DPPTRF('U', nb, AP, info)
+   call DPFTRF('N','U', nb, AP, info)
```
Comment (stefanhenneking) [inline trunk/src/modules/stc.F90:369]:
```suggestion
   call DPFTRF('N','U', Nb, AP, info)
```

**Review by stefanhenneking — APPROVED** [review]
Nice! @jbadger95 do you remember what the performance gain was for high-order elements with RFP over packed format?

With the other minor changes, can you update the revision date of stc_fwd_herm? Thanks!

File: trunk/src/modules/stc.F90 (hunk 1 above)
Comment (jbadger95) [reply trunk/src/modules/stc.F90:348]: ?

Comment (stefanhenneking) [reply trunk/src/modules/stc.F90:348]: These integers were only used in the routine for converting to packed format, as far as I can see.

**Comment by jbadger95** [comment]
@stefanhenneking Done, I changed the base to `next` as well.

**Comment by stefanhenneking** [comment]
Ok, looking to finish up the `update_g/Ddof` PR now (the base changes automatically btw)

**Comment by stefanhenneking** [comment]
LGTM, @jbadger95

**Pull request merged**